api/src/test/java/org/openmrs/module/bedmanagement/validator/BedTagValidatorTest.java

```diff
@@ -0,0 +1,162 @@
+        existing.setName("VIP");
+        bedManagementService.saveBedTag(existing);
+
+        BedTag duplicate = new BedTag();
+        duplicate.setName("VIP");
+
+        Errors errors = new BindException(duplicate, "tag");
+        bedTagValidator.validate(duplicate, errors);
+        assertTrue(errors.hasFieldErrors("name"));
+    }
+
+    @Test
+    public void validate_shouldPassValidationIfExistingTagWithSameNameIsVoided() {
+        BedTag existing = new BedTag();
+        existing.setName("Emergency");
+        existing.setVoided(true);
+        bedManagementService.saveBedTag(existing);
+
+        BedTag newTag = new BedTag();
+        newTag.setName("Emergency");
+
+        Errors errors = new BindException(newTag, "tag");
+        bedTagValidator.validate(newTag, errors);
+        assertFalse(errors.hasFieldErrors("name"));
+    }
+
+    @Test
+    public void validate_shouldFailValidationIfDuplicateNameCaseInsensitive() {
+        BedTag existing = new BedTag();
+        existing.setName("ICU");
+        bedManagementService.saveBedTag(existing);
+
+        BedTag duplicate = new BedTag();
+        duplicate.setName("icu");
+
+        Errors errors = new BindException(duplicate, "tag");
+        bedTagValidator.validate(duplicate, errors);
+        assertTrue(errors.hasFieldErrors("name"));
+    }
+
+    @Test
+    public void validate_shouldPassValidationIfAllRequiredFieldsHaveProperValues() {
+        BedTag tag = new BedTag();
+        tag.setName("Isolation");
+
+        Errors errors = new BindException(tag, "tag");
+        bedTagValidator.validate(tag, errors);
+        assertFalse(errors.hasErrors());
+
+        bedManagementService.saveBedTag(tag);
+        List<BedTag> tags = bedManagementService.getAllBedTags();
+        assertTrue(tags.stream().anyMatch(t -> "Isolation".equals(t.getName())));
+    }
+
+    @Test
+    public void validate_shouldFailValidationIfFieldLengthIsTooLong() {
+        BedTag tag = new BedTag();
+        tag.setName("This name is far too long for a bed tag and should definitely fail validation");
+
+        Errors errors = new BindException(tag, "tag");
+        bedTagValidator.validate(tag, errors);
+        assertTrue(errors.hasFieldErrors("name"));
+    }
+
+    @Test
+    public void validate_shouldPassValidationIfFieldLengthIsWithinLimit() {
+        BedTag tag = new BedTag();
+        tag.setName("General Ward");
+
+        Errors errors = new BindException(tag, "tag");
+        bedTagValidator.validate(tag, errors);
+        assertFalse(errors.hasErrors());
+    }
+
+    @Test
+    public void saveBedTag_shouldNotSaveInvalidTag() {
+        BedTag invalidTag = new BedTag();
+        invalidTag.setName("");
+
+        try {
+            bedManagementService.saveBedTag(invalidTag);
+        } catch (Exception ignored) {
+        }
+
+        List<BedTag> tags = bedManagementService.getAllBedTags();
+        assertFalse(tags.stream().anyMatch(t -> "".equals(t.getName())));
+    }
```

Comment:
We can simplify this test:


```suggestion
	@Test(expected = ValidationException.class)
	public void saveBedTag_shouldNotSaveInvalidTag() {
		BedTag invalidTag = new BedTag();
		invalidTag.setName("");
        bedManagementService.saveBedTag(invalidTag);
	}
```

Reply:
@wikumChamith Thanks for the review. I have fixed the proposed changes. Kindly feel free to review my PR at your convinient time please!